src/hotspot/share/code/nmethod.hpp

```diff
@@ -273,17 +262,10 @@ class nmethod : public CompiledMethod {
   bool _has_flushed_dependencies;      // Used for maintenance of dependencies (CodeCache_lock)
 
   // used by jvmti to track if an event has been posted for this nmethod.
-  bool _unload_reported;
   bool _load_reported;
 
   // Protected by CompiledMethod_lock
-  volatile signed char _state;         // {not_installed, in_use, not_entrant, zombie, unloaded}
-
-#ifdef ASSERT
-  bool _oops_are_stale;  // indicates that it's no longer safe to access oops section
-#endif
-
-  friend class nmethodLocker;
+  volatile signed char _state;         // {not_installed, in_use, not_entrant}
```

Comment:
There's a fourth state: not_used which you might as well add to the comment ?

Reply:
Fixed!